HiP-DataStore/Controllers/TagsController.cs

```diff
@@ -0,0 +1,197 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using PaderbornUniversity.SILab.Hip.DataStore.Core;
+using PaderbornUniversity.SILab.Hip.DataStore.Core.ReadModel;
+using PaderbornUniversity.SILab.Hip.DataStore.Core.WriteModel;
+using PaderbornUniversity.SILab.Hip.DataStore.Model.Entity;
+using PaderbornUniversity.SILab.Hip.DataStore.Model.Rest;
+using PaderbornUniversity.SILab.Hip.DataStore.Model.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MongoDB.Driver;
+using Tag = PaderbornUniversity.SILab.Hip.DataStore.Model.Entity.Tag;
+using PaderbornUniversity.SILab.Hip.DataStore.Model;
+
+namespace PaderbornUniversity.SILab.Hip.DataStore.Controllers
+{
+    [Route("api/[controller]")]
+    public class TagsController : Controller
+    {
+        private readonly EventStoreClient _ev;
+        private readonly CacheDatabaseManager _db;
+        private readonly EntityIndex _entityIndex;
+        private readonly MediaIndex _mediaIndex;
+        private readonly ReferencesIndex _referencesIndex;
+
+        public TagsController(EventStoreClient eventStore, CacheDatabaseManager db, IEnumerable<IDomainIndex> indices)
+        {
+            _ev = eventStore;
+            _db = db;
+            _entityIndex = indices.OfType<EntityIndex>().First();
+            _mediaIndex = indices.OfType<MediaIndex>().First();
+            _referencesIndex = indices.OfType<ReferencesIndex>().First();
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(int), 200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(422)]
+        [ProducesResponseType(409)]
```

Comment:
Method should return 409 if a tag with the given name already exists. This does not seem to be implemented yet.